common/tools/dev-tool/register.js

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * This file loads our monkey patch for module resolution in samples and
+ * tests. We use this to make sure that TS/ES loading is supported and that
+ * path-mapped imports work without rewriting them.
+ *
+ * As a forewarning to anyone who comes to this module and thinks: "Why can't
```

Comment:
I just wanted to say that I **love** this comment.  Thank you so much for explaining why we didn't use TypeScript here and gently warning folks that simply trying to move this to TypeScript may be hard.